### PR TITLE
fix(STONEINTG-583): add status.conditions to existing ITS

### PR DESCRIPTION
**Pull request by dirgim**
* If the ITS was already migrated from v1alpha1, it might not have its status.conditions set at all. This change adds it as part of the scenario controller reconciliation

## Maintainers will complete the following section

- [ ] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [ ] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

File: controllers/scenario/scenario_adapter.go [hunk 1]
```diff
@@ -58,7 +58,19 @@ func NewAdapter(application *applicationapiv1alpha1.Application, scenario *v1bet
...
 
-	// First check if application exists or not
+	// First check if status conditions are set to prevent a known issue with incorrectly migrated v1alpha1 scenarios
+	if a.scenario.Status.Conditions == nil {
+		a.logger.Info("The scenario doesn't have status.conditions set correctly, adding them.")
+		patch := client.MergeFrom(a.scenario.DeepCopy())
+		a.scenario.Status.Conditions = []metav1.Condition{}
+		err := a.client.Status().Patch(a.context, a.scenario, patch)
+		if err != nil {
+			a.logger.Error(err, "Failed to update Scenario")
```
Comment (MartinBasti) [inline controllers/scenario/scenario_adapter.go:68]:
```suggestion
			a.logger.Error(err, "Failed to add Scenario status condition")
```

Comment (dirgim) [reply controllers/scenario/scenario_adapter.go]: Ack, updated.

File: controllers/scenario/scenario_adapter.go [hunk 2]
```diff
@@ -58,7 +58,19 @@ func NewAdapter(application *applicationapiv1alpha1.Application, scenario *v1bet
 // in case it is, set its owner reference
 func (a *Adapter) EnsureCreatedScenarioIsValid() (controller.OperationResult, error) {
 
-	// First check if application exists or not
+	// First check if status conditions are set to prevent a known issue with incorrectly migrated v1alpha1 scenarios
+	if a.scenario.Status.Conditions == nil {
```
Comment (dheerajodha) [inline controllers/scenario/scenario_adapter.go:62]: Are we sure that currently-present-scenarios with this issue, at least have the `.status` field defined?

Comment (dirgim) [reply controllers/scenario/scenario_adapter.go:62]: Yes, status is a subresource so it always gets created, but it's empty for migrated resources.
Here's an example from the staging cluster:
image

Comment (dheerajodha) [reply controllers/scenario/scenario_adapter.go:62]: Thanks for the explanation and screenshot!

**Review by dheerajodha — APPROVED** [review]

**Pull request merged**